lib/pbench/server/indexing_tarballs.py

```diff
@@ -345,6 +340,18 @@ def sighup_handler(*args):
                                 idxctx.logger.warn(
                                     "Unable to advance dataset state: {}", str(e)
                                 )
+                            else:
+                                username = dataset.owner
```

Comment:
Does the code after this point do the right thing if there is an exception raised by `Dataset.attach()`?  I imagine that `username` reads as `None` (which is probably OK), but what does it mean for `dataset` to be undefined?

Reply:
This then begs the question of supporting no-owner tar balls.  Is this something we want to continue to support with this commit?  I don't remember what was decided about that.